app/models/user.rb

```diff
@@ -7,6 +7,7 @@ class User < ActiveRecord::Base
   validates :name, presence: true
   validates :email, presence: true, uniqueness: true
   validates :password, presence: true, on: :create, length: { minimum: 7 }
+  before_create :generate_token
```

Comment:
I'm not totally following how this is better, but I don't like generating tokens for all the users in the db with a migration. Plus, it's really only used when there's a password reset. Thus, this field could be null until a user tries to reset the password. Perhaps, a expiry on the token would be nice in case someone compromises the db, even with an email address, then tokens would expire anyway